Client/AccessoryViewProvider.swift

```diff
@@ -0,0 +1,144 @@
+        NSLayoutConstraint.activate([
+            view.widthAnchor.constraint(equalToConstant: 2),
+            view.heightAnchor.constraint(equalToConstant: 30)
+        ])
+
+        view.accessibilityElementsHidden = true
+    }
+
+    lazy private var cardImageView: UIImageView = .build { imageView in
+        imageView.image = UIImage(named: ImageIdentifiers.creditCardPlaceholder)?.withRenderingMode(.alwaysTemplate)
+        imageView.contentMode = .scaleAspectFit
+        imageView.accessibilityElementsHidden = true
+
+        NSLayoutConstraint.activate([
+            imageView.widthAnchor.constraint(equalToConstant: 24),
+            imageView.heightAnchor.constraint(equalToConstant: 24)
+        ])
+    }
+
+    lazy private var useCardTextLabel: UILabel = .build { label in
+        label.font = DynamicFontHelper.defaultHelper.preferredFont(withTextStyle: .title3, size: 16, weight: .medium)
+        label.text = .CreditCard.Settings.UseSavedCardFromKeyboard
+        label.numberOfLines = 0
+    }
+
+    private lazy var cardButtonStackView: UIStackView = .build { [weak self] stackView in
+        guard let self = self else { return }
+
+        stackView.addArrangedSubview(self.fixedSpacer)
+        stackView.addArrangedSubview(self.cardImageView)
+        stackView.addArrangedSubview(self.useCardTextLabel)
+        stackView.addArrangedSubview(self.fixedSpacer)
+        stackView.spacing = 2
+        stackView.distribution = .equalCentering
+    }
+
+    init(for target: TabWebView,
+         themeManager: ThemeManager = AppContainer.shared.resolve(),
+         notificationCenter: NotificationCenter = NotificationCenter.default) {
+        self.target = target
+        self.themeManager = themeManager
+        self.notificationCenter = notificationCenter
+
+        super.init(frame: CGRect(width: UIScreen.main.bounds.width,
+                                 height: AccessoryViewUX.toolbarHeight))
+
+        listenForThemeChange(self)
+        setupLayout()
+        applyTheme()
+
+        self.notificationCenter.addObserver(self,
+                                            selector: #selector(reloadViewWithDelay),
+                                            name: .CreditCardAccessoryNeeded,
+                                            object: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    @objc
+    private func reloadViewWithDelay() {
+        throttler.throttle { [weak self] in
+            guard let self = self else { return }
+
+            self.showCreditCard = true
+
+            self.setNeedsLayout()
+            self.setupLayout()
+            self.layoutIfNeeded()
+        }
+    }
+
+    private func setupLayout() {
+        translatesAutoresizingMaskIntoConstraints = false
+
+        if showCreditCard {
+            let cardStackViewForBarButton = UIBarButtonItem(customView: cardButtonStackView)
```

Comment:
@nishant2718 
Would recommend adding an action stub for action for this button in the form of a closure / delegate that can be configured to perform a task outside the view as it will be used to show the bottom sheet in future